src/Library/demos/Level Bars/main.js

```diff
@@ -0,0 +1,82 @@
+import Gtk from "gi://Gtk";
+
+const bar_continuous = workbench.builder.get_object("bar_continuous");
+const percentage_label = workbench.builder.get_object("percentage_label");
+
+bar_continuous.add_offset_value("full", 100);
+bar_continuous.add_offset_value("half", 50);
+bar_continuous.add_offset_value("low", 25);
+
+bar_continuous.connect("notify::value", () => {
+  percentage_label.label = `${Math.trunc(bar_continuous.value)}%`;
+});
+
+const bar_discrete = workbench.builder.get_object("bar_discrete");
+
+bar_discrete.add_offset_value("very-weak", 1);
+bar_discrete.add_offset_value("weak", 2);
+bar_discrete.add_offset_value("moderate", 4);
+bar_discrete.add_offset_value("strong", 6);
+
+const entry = workbench.builder.get_object("entry");
+const label_strength = workbench.builder.get_object("label_strength");
+
+/*
+Rules for password strength
+Very Weak- 1-3 characters long, doesn't use all valid characters
+Weak     - >=4 characters long, doesn't use all valid characters
+Moderate - 5-7 characters long, uses all valid characters once
+Strong   - >=8 characters long, uses all valid characters once
+
+Valid characters include uppercase and lowercase letters,
+special characters and digits
+*/
+
+const evaluate_strength = [
```

Comment:
This is too complicated for a demo of Gtk.LevelBar
I think using password strength for the demo is a good idea, use the password length only :pray: 

Also https://xkcd.com/936/ :D
In general – better to use ready-made solutions for this kind of security sensitive things. For example https://github.com/dropbox/zxcvbn

Reply:
I forgot about that xkcd comic :)